### ISSUE TITLE
Propose some API changes to this package

### ISSUE THREAD

**Issue by ocefpaf** [dnowacki-usgs/nwis#1]
`nwis` has almost everything I like in a package:

- single purpose
- simple and direct
- well written and easy to read/hack on it

However, the name of the function, `nwis_json`, has json on it but the default return is a pandas DataFrame. There is also an option to change that to a xarray Dataset. That is quite confusing and I would like to propose to return only the JSON and add to extra functions to perform the conversion. To be 100% honest I would be OK with only the json return I would deal with the conversions myself but I can see how users will want that build in.

What do you think @dnowacki-usgs?
I think it is an overkill but I would be OK with a class that has the json and the two methods to convert.

**Comment by dnowacki-usgs** [comment]
I agree, the function name is not ideal; things have evolved as I have found my uses for this tool changing. So you are proposing that the tool return the JSON in a class with methods that will convert and return that data as pandas or xarray?

**Comment by ocefpaf** [comment]
> So you are proposing that the tool return the JSON in a class with methods that will convert and return that data as pandas or xarray?

That is one idea but not my first choice. I would prefer it returns just the JSON and then two extra functions that handles the conversions.

**Comment by rsignell-usgs** [comment]
@ocefpaf, I like this idea.   Keep it simple and transparent!

**Comment by dnowacki-usgs** [comment]
I created two branches, one with a [class approach](https://github.com/dnowacki-usgs/nwis/tree/nwisclass) and one with a [functional approach](https://github.com/dnowacki-usgs/nwis/tree/nwisfunc). Let me know if either of these is anything close to what you had in mind.
